drivers/sensor/qdec_s32/qdec_s32.c

```diff
@@ -0,0 +1,474 @@
+		/* Lcu_Ip_HwOutputStateType ** ppxHwOutputStateArray;			    */	\
+		&Lcu_Ip_ppxHwOutputStateArray_Config[0],					\
+		/* const Lcu_Ip_LogicInstanceType * const * ppxLogicInstanceConfigArray     */	\
+		&Lcu_Ip_ppxLogicInstanceArray_Config[0],					\
+		/* const Lcu_Ip_LogicOutputConfigType * const pxLogicOutputResetConfigArray */	\
+		&Lcu_Ip_LogicOutputResetConfig,							\
+		/* const Lcu_Ip_LogicInputConfigType * const pxLogicInputResetConfigArray   */	\
+		&Lcu_Ip_LogicInputResetConfig,							\
+		/* const Lcu_Ip_LogicOutputType * const * ppxLogicOutputConfigArray	    */	\
+		&Lcu_Ip_ppxLogicOutputArray_Config[0],						\
+		/* const Lcu_Ip_LogicInputType * const * ppxLogicInputConfigArray	    */	\
+		&Lcu_Ip_ppxLogicInputArray_Config[0],						\
+	};
+
+#define Trgmux_Ip_LogicTrigger_Config(n, LogicChannel, Output, Input)				\
+	{											\
+		/* uint8 LogicChannel; */ LogicChannel,						\
+		/* uint8 Output; */ Output,							\
+		/* uint8 Input; */ Input,							\
+		/* uint8 HwInstId; */ DT_INST_PROP(n, trgmux_inst),				\
+		/* boolean Lock; */ (boolean)FALSE,						\
+	};
+
+#define TRGMUX_IP_INIT_CONFIG(n)								\
+	const Trgmux_Ip_LogicTriggerType Trgmux_Ip_LogicTrigger_0_Config =			\
+		Trgmux_Ip_LogicTrigger_Config(							\
+		n, DT_INST_PROP_BY_IDX(n, trgmux_io_config, 0),				\
+		DT_INST_PROP_BY_IDX(n, trgmux_io_config, 1),				\
+		DT_INST_PROP_BY_IDX(n, trgmux_io_config,				\
+				    2)) const Trgmux_Ip_LogicTriggerType		\
+		Trgmux_Ip_LogicTrigger_1_Config = Trgmux_Ip_LogicTrigger_Config(	\
+			n, DT_INST_PROP_BY_IDX(n, trgmux_io_config, 3),			\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 4),			\
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config,			\
+					    5)) const Trgmux_Ip_LogicTriggerType	\
+		Trgmux_Ip_LogicTrigger_2_Config = Trgmux_Ip_LogicTrigger_Config(	 \
+			n, DT_INST_PROP_BY_IDX(n, trgmux_io_config, 6),			 \
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 7),			 \
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config,			 \
+					    8)) const Trgmux_Ip_LogicTriggerType	 \
+		Trgmux_Ip_LogicTrigger_3_Config = Trgmux_Ip_LogicTrigger_Config( \
+			n, DT_INST_PROP_BY_IDX(n, trgmux_io_config, 9),		 \
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 10),		 \
+			DT_INST_PROP_BY_IDX(n, trgmux_io_config, 11))		 \
+			const Trgmux_Ip_InitType Trgmux_Ip_Init_Config = {	 \
+				{						 \
+					&Trgmux_Ip_LogicTrigger_0_Config,	 \
+					&Trgmux_Ip_LogicTrigger_1_Config,	 \
+					&Trgmux_Ip_LogicTrigger_2_Config,	 \
+					&Trgmux_Ip_LogicTrigger_3_Config,	 \
+				},						 \
+	};
+
+#define eMios_Icu_Ip_ChannelConfig_Common(channel)						\
+	{											\
+		channel, EMIOS_ICU_SAIC, (boolean)FALSE, EMIOS_PRESCALER_DIVIDE_1,		\
+			EMIOS_PRESCALER_DIVIDE_1, EMIOS_ICU_BUS_INTERNAL_COUNTER,		\
+			EMIOS_ICU_MODE_EDGE_COUNTER, EMIOS_ICU_MODE_WITHOUT_DMA,		\
+			EMIOS_ICU_NO_MEASUREMENT, EMIOS_ICU_RISING_EDGE,			\
+			EMIOS_DIGITAL_FILTER_BYPASSED, NULL_PTR, NULL_PTR, (uint8)255U,		\
+			(boolean)FALSE, .eMiosChannelNotification = NULL_PTR,			\
```

Comment:
Is this possible to configure overflow callback here? instead of touching on HAL's internal state like you did above: https://github.com/zephyrproject-rtos/zephyr/blob/f2e74229013c15f579854e4be61dfd3da16b18f4/drivers/sensor/qdec_s32/qdec_s32.c#L199-L202

Reply:
@Dat-NguyenDuy .. Assigning anything to these structure members will only be effective if I can pass -  eMios_Icu_Ip_ConfigType **eMios_Icu_Ip_Config** to Emios_Icu_Ip_Init. But since I cannot call it so I need to either call Emios exposed API's or use these global variables.
But as discussed we will fix it in a follow up PR